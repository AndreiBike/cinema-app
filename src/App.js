import React from 'react';

// Creating element using React.Component
class App extends React.Component {

    render () {
        return (
            // Creating element using React.createElement
            React.createElement("div",null, <h1>Hallo world (React.CreateElement)</h1>, <PureComp/>, <FuncComp />) 
        )
    }
}

// Creating element using React.PureComponent
class PureComp extends React.PureComponent {

    render() {
        return (
            <div>
                <h1>Hallo world (React.PureComponent)</h1>
            </div>
        )
    }
}

// Creating element using functional component
const FuncComp = (props) => {
    return (
        <h1>Hallo world (FunctionalComponent)</h1>
    )
}


export default App;