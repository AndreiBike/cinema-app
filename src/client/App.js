import React from 'react';
import { Provider } from 'react-redux';
import reduxState from '@root/client/reduxStore/reduxStore';
import Footer from '@root/client/components/footer/Footer';
import Header from '@root/client/components/header/Header';
import Sortbar from '@root/client/components/sortbar/Sortbar';
import Movies from '@root/client/components/movies/Movies';
import ErrorBoundary from '@root/client/components/errorBoundary/ErrorBoundary';
import Description from '@root/client/components/description/Description';
import './App.module.css';

const headerLabels = {
  netflixRoulette: ["NETFLIX", "ROULETTE"],
  findYourMovie: "FIND YOUR MOVIE",
};

const footerLabels = {
  netflixRoulette: ["NETFLIX", "ROULETTE"],
}


const App = (props) => {
  return (
    <Provider store={reduxState}>
      <div className="app">
        <ErrorBoundary>
          <Header headerLabels={headerLabels} />
         {/* <Description headerLabels={headerLabels} /> */}
          <Sortbar />
          <Movies />
          <Footer footerLabels={footerLabels} />
        </ErrorBoundary>
      </div>
    </Provider>
  )
}

export default App;
