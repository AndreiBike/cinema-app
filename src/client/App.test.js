import React from 'react';
import {configure, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import App from "./App";

configure({adapter: new Adapter()});

describe("<App> component testing", () => {
  const component = mount(<App />);

  it ("+++capturing snapshot of App", ()=> {
    expect(component).toMatchSnapshot();
  })
})