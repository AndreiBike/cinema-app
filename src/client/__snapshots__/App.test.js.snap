// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App> component testing +++capturing snapshot of App 1`] = `ReactWrapper {}`;
